qiskit_nature/properties/grouped_property.py

```diff
@@ -12,11 +12,16 @@
 
 """A group of multiple properties."""
 
+from __future__ import annotations
+
 from collections.abc import Iterable
 from typing import Dict, Generator, Generic, Optional, Type, TypeVar, Union
 
+import h5py
+
+from qiskit_nature.hdf5 import _import_and_build_from_hdf5
```

Comment:
This is the only exception where this method is being used. I do not think it is worth adding that method to the public API for that.

Reply:
I think it can be left out of the `all` so that you have to know and to explicitly import it. Across pkgs like this I know I am ending up looking at 'accessing private member` highlights in my IDE as its warning about that. I used to see this with the builders that were marked private yet used by other pkgs. Clearly the method then has legitimate use here outside of the pkg so its not truly private in that regard - maybe you're more trying to dissuade others from using this outside the pkg by annotating it private despite the fact its done here.

Reply:
Right now I did not plan on exposing any of the `hdf5` methods in the `__all__` as this would be the root of Qiskit Nature. Instead I think it makes sense to simply have to import them from `qiskit_nature.hdf5`. Since this is a single file the import path will always be that.

> maybe you're more trying to dissuade others from using this outside the pkg by annotating it private despite the fact its done here

Yes this is was my reasoning :+1: